### PR TITLE
[BugFix] Fix some query exceptions of hudi table

### PR DESCRIPTION
**Pull request by miomiocat**
## What type of PR is this：
- [x] BugFix
- [ ] Feature
- [ ] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

## Which issues of this PR fixes ：
- fix refresh exception
- fix query exception for empty table
- add stack trace log

### CODE REVIEW

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 1 / 15 (06.67%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/catalog/HudiTable.java|0|5|00.00%|[250, 251, 252, 253, 254]|
|:large_blue_circle:|com/starrocks/external/hudi/HudiRemoteFileIO.java|1|10|10.00%|[53, 61, 62, 63, 64, 65, 68, 72, 85]|

File: fe/fe-core/src/main/java/com/starrocks/catalog/HudiTable.java [hunk 1]
```diff
@@ -245,8 +247,12 @@ public TTableDescriptor toThrift(List<DescriptorTable.ReferencedPartitionInfo> p
         Configuration conf = new Configuration();
         HoodieTableMetaClient metaClient =
                 HoodieTableMetaClient.builder().setConf(conf).setBasePath(getTableLocation()).build();
-        HoodieTimeline timeline = metaClient.getActiveTimeline().filterCompletedAndCompactionInstants();
-        String queryInstant = timeline.lastInstant().get().getTimestamp();
+        HoodieTimeline timeline = metaClient.getCommitsAndCompactionTimeline().filterCompletedInstants();
+        Option<HoodieInstant> latestInstant = timeline.lastInstant();
+        String queryInstant = "";
+        if (latestInstant.isPresent()) {
```
Comment (DorianZheng) [inline fe/fe-core/src/main/java/com/starrocks/catalog/HudiTable.java:253]: when does this condition not establish?

**Review by DorianZheng — DISMISSED** [review]
LGTM

**Review by stephen-shelby — DISMISSED** [review]
LGTM

**Review by Youngwb — DISMISSED** [review]

**Comment by wanpengfei-git** [comment]
run starrocks_admit_test

**Review by dirtysalt — DISMISSED** [review]

**Comment by wanpengfei-git** [comment]
run starrocks_admit_test

File: fe/fe-core/src/main/java/com/starrocks/catalog/HudiTable.java (hunk 1 above)
Comment (miomiocat) [reply fe/fe-core/src/main/java/com/starrocks/catalog/HudiTable.java:253]: newly created empty table

**Review by dirtysalt — APPROVED** [review]

**Review by Youngwb — APPROVED** [review]

**Comment by miomiocat** [comment]
run starrocks_fe_unittest

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 0 / 14 (00.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/catalog/HudiTable.java|0|5|00.00%|[250, 251, 252, 253, 254]|
|:large_blue_circle:|com/starrocks/external/hudi/HudiRemoteFileIO.java|0|9|00.00%|[53, 61, 62, 63, 64, 65, 68, 72, 85]|

**Pull request merged**